File: Data Structures & Algorithms/jump-game-ii/submission-2.cpp
class Solution {
public:
    int jump(vector<int>& nums) {
        int l = 0, r = 0;
        int count = 0;
        int far = 0;
        while( r < nums.size()-1){
            for(int i = l; i <= r; i++){
                far= max(far, nums[i]+i);
            }

            l = r+1;
            r = far;
            count ++;
        }
        return count;
    }
};

File: Data Structures & Algorithms/majority-element-ii/submission-2.py
class Solution:
    def majorityElement(self, nums: List[int]) -> List[int]:
        count = defaultdict(int)
        res = []
        for i in range(len(nums)):
            count[nums[i]]+=1
            if len(count) > 2:
                newcount = defaultdict(int)
                for k, v in count.items():
                    if v > 1:
                        newcount[k] = v - 1
                count = newcount

        for k,v in count.items():
            if nums.count(k) > len(nums) // 3:
                res.append(k)
        return res